src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs

```diff
@@ -27,6 +61,58 @@ public TypeSyntaxSimplifierWalker(CSharpSimplifyTypeNamesDiagnosticAnalyzer anal
             _semanticModel = semanticModel;
             _optionSet = optionSet;
             _cancellationToken = cancellationToken;
+
+            var root = semanticModel.SyntaxTree.GetRoot(cancellationToken);
+            _aliasedNames = GetAliasedNames(root as CompilationUnitSyntax);
```

Comment:
this should be a normal cast.

Reply:
This doesn't seem problematic, and there are known cases where syntax trees do not have `CompilationUnitSyntax` as the root.

Reply:
> and there are known cases where syntax trees do not have CompilationUnitSyntax as the root.

Those would all be bugs in the case of analyzers though.  No one should ever register a semantic-model analyzer and *not* get a CompilationUnit.  That would be a big invariance bug.

Reply:
The rewrite of GetAliasedNames specifically targets scenarios where the entry point is _not_ a semantic model analyzer. Semantic model analyzers always start at the root so the normal recursive descent walker can be used for this.

Reply:
i'm not certain what you mean.  When is the entrypoint not a semantic-model analyzer?

Reply:
In the current code, it's not. But we're considering a hybrid code block + syntax node analyzer if we can't get the performance up through SemanticModel sharing in the compiler or driver.